src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -3067,6 +3044,19 @@ void ShenandoahHeap::entry_updaterefs() {
   op_updaterefs();
 }
 
+void ShenandoahHeap::entry_weak_refs() {
+  static const char* msg = "Concurrent weak references";
+  ShenandoahConcurrentPhase gc_phase(msg, ShenandoahPhaseTimings::conc_weak_refs);
+  EventMark em("%s", msg);
+
+  ShenandoahWorkerScope scope(workers(),
+                              ShenandoahWorkerPolicy::calc_workers_for_conc_root_processing(),
```

Comment:
It probably does not matter, but maybe we should be having a separate `ShenandoahWorkerPolicy` entry for `conc_weak_refs`.